Open Judge System/OJS.Common/Extensions/FileHelpers.cs

```diff
@@ -101,6 +101,27 @@ public static void DeleteFiles(params string[] filePaths)
             }
         }
 
+        public static string ExtractFileFromZip(string pathToArchive, string fileName, string destinationDirectory)
+        {
+            using (var zip = new ZipFile(pathToArchive))
+            {
+                ZipEntry entryToExtract = zip.Entries.FirstOrDefault(f => f.FileName.EndsWith(fileName));
+                if (entryToExtract == null)
+                {
+                    throw new FileNotFoundException($"{fileName} not found in submission!");
+                }
+
+                entryToExtract.Extract(destinationDirectory);
+
+                string extractedFilePath =
+                    Directory.EnumerateFiles(destinationDirectory, fileName, SearchOption.AllDirectories)
```

Comment:
Is there really a need to enumerate through the folders again, if you need the path cant you get it from concatenating the destinationDirectory with the entry's relative path.